vendor/bat-native-ads/src/bat/ads/internal/resources/behavioral/purchase_intent/purchase_intent_resource.h

```diff
@@ -6,15 +6,19 @@
 #ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_RESOURCES_BEHAVIORAL_PURCHASE_INTENT_PURCHASE_INTENT_RESOURCE_H_
 #define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_RESOURCES_BEHAVIORAL_PURCHASE_INTENT_PURCHASE_INTENT_RESOURCE_H_
 
+#include <memory>
 #include <string>
 
+#include "base/memory/weak_ptr.h"
 #include "bat/ads/internal/ad_targeting/data_types/behavioral/purchase_intent/purchase_intent_info.h"
+#include "bat/ads/internal/resources/parsing_result.h"
 #include "bat/ads/internal/resources/resource.h"
 
 namespace ads {
 namespace resource {
 
-class PurchaseIntent final : public Resource<ad_targeting::PurchaseIntentInfo> {
+class PurchaseIntent final
+    : public Resource<const ad_targeting::PurchaseIntentInfo*> {
```

Comment:
Also all `<something>_resource.cc` looks almost identical:
same code, `is_initialized_/unique_ptr/WeakPtrFactory` fields.

Could we consider to merge it in one class (maybe template) to avoid code duplication? (in another issue)